app/models/concerns/foreman/sti.rb

```diff
@@ -8,11 +8,11 @@ class << base
 
     module ClassMethods
       # ensures that the correct STI object is created when :type is passed.
-      def new(*attributes, &block)
-        if (h = attributes.first).is_a?(Hash) && (type = h.with_indifferent_access.delete(:type)) && !type.empty?
+      def new(attributes = nil, &block)
+        if attributes.is_a?(Hash) && (type = attributes.with_indifferent_access.delete(:type)) && !type.empty?
```

Comment:
not blocking, especially as this is old code, but why the `.delete` here? is that to emulate a `.fetch(:type, nil)` or a `.dig(:type)? (if I read Rals code correctly, `with_indifferent_access` returns a copy, so `attributes` is never modified?)

Reply:
I guess so, didn't want to change all that much just to be extra safe.